docs/docs/reference/contextual/instance-defs.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: doc-page
+title: "Implied Instances"
+---
+
+Implied instance definitions define "canonical" values of given types
+that serve for synthesizing arguments to [inferable parameters](./inferable-params.html). Example:
+
+```scala
+trait Ord[T] {
+  def compare(x: T, y: T): Int
+  def (x: T) < (y: T) = compare(x, y) < 0
+  def (x: T) > (y: T) = compare(x, y) > 0
+}
+
+implied IntOrd for Ord[Int] {
+  def compare(x: Int, y: Int) =
+    if (x < y) -1 else if (x > y) +1 else 0
+}
+
+implied ListOrd[T] given (ord: Ord[T]) for Ord[List[T]] {
+  def compare(xs: List[T], ys: List[T]): Int = (xs, ys) match {
+    case (Nil, Nil) => 0
+    case (Nil, _) => -1
+    case (_, Nil) => +1
+    case (x :: xs1, y :: ys1) =>
+      val fst = ord.compare(x, y)
+      if (fst != 0) fst else xs1.compareTo(ys1)
+  }
+}
+```
+This code defines a trait `Ord` and two implied instance definitions. `IntOrd` defines
+an implied instance for the type `Ord[Int]` whereas `ListOrd` defines implied
+instances of `Ord[List[T]]` for all types `T` that come with an implied `Ord` instance themselves.
+The `given` clause in `ListOrd`  defines an [inferable parameter](./inferable-params.html).
+Inferable parameters are further explained in the next section.
+
+## Anonymous Implied Instances
+
+The name of an implied instance can be left out. So the implied instance definitions
+of the last section can also be expressed like this:
+```scala
+implied for Ord[Int] { ... }
+implied [T] given (ord: Ord[T]) for Ord[List[T]] { ... }
+```
+If the name of an instance is missing, the compiler will synthesize a name from
+the type(s) in the `for` clause.
+
+## Implied Alias Instances
+
+An implied alias instance defines an implied instance that is equal to some expression. E.g.,
+```scala
+implied ctx for ExecutionContext = currentThreadPool().context
```

Comment:
I fail to see what is going on in this example. What is `currentThreadPool()`? At what scope it resides? I think a fuller example is required here.

Reply:
It's just an example I think, like `implied ctx for ExecutionContext = ???`.

Reply:
I guess @soronpo meant that it would be useful to use a very commonly known pattern to make the intention of the example clearer.
People might be mislead by trying to understand what the example is actually trying to achieve...

something more common might be
```scala
implied ctx for ExecutionContext = myActorSystem.dispatcher
```
?

Reply:
Ok, but for me as a not frequenty Akka user `currentThreadPool().context` is certainly more intuitively understandable than `myActorSystem.dispatcher`

Reply:
not if `currentThreadPool` doesn't exist in any known library...

but my example was just out-of-the-blue, something more common ćould be found